Blizzard.IO.RabbitMQ.Builders/Rpc/BaseNetqRabbitRpcBuilder.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Blizzard.IO.Core.Rpc;
+using System.Collections.Generic;
+using Blizzard.IO.RabbitMQ.Rpc;
+using System;
+using Blizzard.IO.RabbitMQ.Entities.Rpc;
+using Microsoft.Extensions.Logging;
+using Blizzard.IO.Serialization.Rpc;
+
+namespace Blizzard.IO.RabbitMQ.Builders.Rpc
+{
+    public class BaseNetqRabbitRpcBuilder
+    {
+        private static Dictionary<RpcConnectionKey, INetqRabbitRpcConnection<Func<Type, object>>> netqRpcConnections = new Dictionary<RpcConnectionKey, INetqRabbitRpcConnection<Func<Type, object>>>();
+
+        protected ISerializer Serializer = new JsonSerializer();
+        protected string Hostname = "localhost";
+        protected string Password = "guest";
+        protected string Username = "guest";
+        protected RpcConfiguration RpcConfiguration = new RpcConfiguration
+        {
+            ExchangeNameProvider = type => "RPC_EXCHANGE",
+            RoutingKeyProvider = type => type.ToString()
+        };
+        protected RpcMessageType RpcMessageType = RpcMessageType.Concrete;
+        protected ushort Timeout = 10;
+        protected string Product = null;
+        protected string Platform = null;
+        protected string VirtualHost = null;
```

Comment:
```suggestion
        protected string Product = "";
        protected string Platform = "";
        protected string VirtualHost = "/";
```

Reply:
I want to use the default one that easynetq offer in the platform and the product I dont them to just be empty

Reply:
If you look at the extension function when the strings are empty it dosen't add them to the connectionString and then it will use the default.
If you want it to be like this then the extension check them for null and not for empty string , maybe i should change it. Tell me what you think

Reply:
The problem is that im not using the extension function cause I need to initialize the bus in a more advance way